src/dialect/postgresql.rs

```diff
@@ -20,7 +20,7 @@ impl Dialect for PostgreSqlDialect {
         // See https://www.postgresql.org/docs/11/sql-syntax-lexical.html#SQL-SYNTAX-IDENTIFIERS
         // We don't yet support identifiers beginning with "letters with
         // diacritical marks and non-Latin letters"
-        ('a'..='z').contains(&ch) || ('A'..='Z').contains(&ch) || ch == '_'
+        ('a'..='z').contains(&ch) || ('A'..='Z').contains(&ch) || ch == '_' || ch == '#'
```

Comment:
Yeah I am torn on this one - I worry about causing trouble for someone who is relaying on `#` not being a valid pg identifiers. I don't really have a great understanding of what people are doing / using sqlparser for and I would say it is not particularly consistent about disallowing constructs that are not valid other dialects of SQL 🤔 

What do you think about adding a `RedshiftSqlDialect` to avoid unintended consquences? I think it could be pretty straightforward, something like:

```rust
impl Dialect for RedshiftSqlDialect {
    fn is_identifier_start(&self, ch: char) -> bool {
      PostgreSqlDialect::is_identifier_start(ch) || ch == '#'
    }
}
```